### PR TITLE
docs: ImageCache

### PR DESCRIPTION
**Pull request by Ombuweb**

### CODE REVIEW

**Comment by Ombuweb** [comment]
When is the `ImageCache.downloadedEvent` triggered? The methods `enableDownload()` and `disableDownload()` don't seem to have any effect.

File: content/guide/ui/image-cache.md [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+    imageCache = new ImageCache()
+    
+    constructor(){
+        super()
+    }
+    
+}
+```
+### Save an image to cache
+To retrive an image from a remote server and cache it, use either the [push()](#push) or [enqueue()](#enqueue) method. These methods cache the image as `android.graphics.Bitamap`(Android) or `UIImage`(iOS) instance.
```
Comment (ashutosh887) [inline content/guide/ui/image-cache.md]: It shold be `android.graphics.Bitmap`

**Review by ashutosh887 — CHANGES_REQUESTED** [review]

File: content/guide/ui/image-cache.md (hunk 1 above)
Comment (NathanWalker) [inline content/guide/ui/image-cache.md]:
```suggestion
To retrieve an image from a remote server and cache it, use either the [push()](#push) or [enqueue()](#enqueue) method. These methods cache the image as `android.graphics.Bitmap` (Android) or `UIImage` (iOS) instance.
```

File: content/guide/ui/image-cache.md [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+:::tip Note
+To call the [push()](#push) method instead, just replace `enqueue` with `push`
+:::
+
+### Getting an image from cache
+To get an image from the cache, use the [get()]() method.
+
+```ts
+const cachedImage = this.imageCache.get("cat")
+if(cacheImage) notifyProperpyChange("imageSource", cachedImage)
```
Comment (NathanWalker) [inline content/guide/ui/image-cache.md]:
```suggestion
// cachedImage for use
```

**Comment by ashutosh887** [comment]
Can we close this now? @NathanWalker

File: content/guide/ui/image-cache.md [hunk 3]
```diff
@@ -0,0 +1,146 @@
+---
+title: ImageCache
+---
+
+The ImageCache class allows you to cache an image from an HTTP request.
```
Comment (NathanWalker) [inline content/guide/ui/image-cache.md]:
```suggestion
ImageCache allows you to cache an image from an HTTP request.
```

File: content/guide/ui/image-cache.md [hunk 4]
```diff
@@ -0,0 +1,146 @@
+---
+title: ImageCache
+---
+
+ImageCache allows you to cache an image from an HTTP request.
+
+## Using ImageCache
+For consistency, you should use a single ImageCache instance
```
Comment (NathanWalker) [inline content/guide/ui/image-cache.md]:
```suggestion
For consistency, you should use a single ImageCache instance throughout your app.
```

File: content/guide/ui/image-cache.md [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+throughout your data model.
```
Comment (NathanWalker) [inline content/guide/ui/image-cache.md]:
```suggestion

```

File: content/guide/ui/image-cache.md [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+```ts
+const cachedImage = this.imageCache.get("cat")
+// cachedImage for use
+```
+
+## ImageCache API
+
+### constructor
+```ts
+imageCache = new ImageCache()
```
Comment (NathanWalker) [inline content/guide/ui/image-cache.md]:
```suggestion
const imageCache = new ImageCache()
```

File: content/guide/ui/image-cache.md [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+
+### push()
+```ts
+this.imageCache.push(request)
+```
+Adds a new image to the **start** of the cache. `request` is an object with the following members:
+
+- `url`(`string`): The url of the image.
+- `key`(`string`): The name to uniquely identify the image in the cache.
+-  _Optional_: `complete` is a function to be called upon caching an image successfully.
```
Comment (NathanWalker) [inline content/guide/ui/image-cache.md]:
```suggestion
-  _Optional_: `completed` is a function to be called upon caching an image successfully. 
```

File: content/guide/ui/image-cache.md [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+```
+Adds a new image to the **end** of the cache. For more details about the `request` parameter, see the [push()](#push) method.
+
+---
+
+## remove()
+```ts
+this.imageCache.remove(imageKey)
+```
+It removes the specified image from the cache.
```
Comment (NathanWalker) [inline content/guide/ui/image-cache.md]:
```suggestion
Remove the specified image from the cache.
```

File: content/guide/ui/image-cache.md [hunk 9]
```diff
@@ -0,0 +1,145 @@
...
+```
+Remove the specified image from the cache.
+
+---
+
+### clear()
+```ts
+this.imageCache.clear()
+```
+It clears all the images from the cache.
```
Comment (NathanWalker) [inline content/guide/ui/image-cache.md]:
```suggestion
Clears all the images from the cache.
```

**Pull request merged**